upstairs/src/test.rs

```diff
@@ -5151,21 +5035,13 @@ pub(crate) mod up_test {
             }],
         }]);
 
-        let result =
-            std::panic::catch_unwind(std::panic::AssertUnwindSafe(|| {
-                ds.process_ds_completion(
-                    next_id,
-                    ClientId::new(0),
-                    response,
-                    &Some(context),
-                    UpState::Active,
-                    None,
-                )
-            }));
-        assert!(result.is_err());
+        let _result = upstairs
+            .process_ds_operation(next_id, ClientId::new(0), response, None)
+            .await;
     }
 
     #[tokio::test]
+    #[should_panic]
```

Comment:
Same `should_panic` here,  If there is any `unwrap()` before the expected panic, I think this 
tests will pass for the wrong reasons.

Reply:
Same, see 707f4d9